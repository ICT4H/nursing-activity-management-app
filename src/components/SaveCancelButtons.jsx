import React, {Component} from 'react';

class SaveCancelButtons extends Component {
  render() {
    return (
        <div>
          <button onClick={this.props.saveFn} className={"button"} value={"Save"}>Save</button>
          <button onClick={this.props.cancelFn} className={"button"} value={"Cancel"}>Cancel</button>
        </div>
    )
  }
}

export default SaveCancelButtons
