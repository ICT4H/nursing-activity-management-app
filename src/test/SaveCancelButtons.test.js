import React from 'react';
import SaveCancelButtons from '../SaveCancelButtons';
import renderer from 'react-test-renderer';


let saveFn = () => true;
let cancelFn = () => true;

test('Should be a div with 2 two buttons', () => {
  let component = renderer.create(<SaveCancelButtons saveFn={saveFn} cancelFn={cancelFn}/>);
  let tree = component.toJSON();
  expect(component.root.findAll(element => element.type === "button")).toHaveLength(2);
  expect(tree).toMatchSnapshot();
});

test('Should be a div with given saveFn as on click of save button && cancelFn as on click of cancel button', () => {
  let component = renderer.create(<SaveCancelButtons saveFn={saveFn} cancelFn={cancelFn}/>);
  let findSaveButton = element => element.type === "button" && element.props.value === "Save";
  let findCancelButton = element => element.type === "button" && element.props.value === "Cancel";
  expect(component.root.find(findSaveButton));
  expect(component.root.find(findCancelButton));
});