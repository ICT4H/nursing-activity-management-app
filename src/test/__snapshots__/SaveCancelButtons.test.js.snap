// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should be a div with 2 two buttons 1`] = `
<div>
  <button
    className="button"
    onClick={[Function]}
    value="Save"
  >
    Save
  </button>
  <button
    className="button"
    onClick={[Function]}
    value="Cancel"
  >
    Cancel
  </button>
</div>
`;
